src/docbuilder/limits.rs

```diff
@@ -109,78 +114,88 @@ mod test {
                 ..defaults
             };
             Overrides::save(
-                &mut db.conn(),
+                &mut conn,
                 krate,
                 Overrides {
                     memory: Some(limits.memory),
                     targets: Some(limits.targets),
                     timeout: Some(limits.timeout),
                 },
-            )?;
+            )
+            .await?;
             assert_eq!(
                 limits,
-                Limits::for_crate(&env.config(), &mut db.conn(), krate)?
+                Limits::for_crate(&env.config(), &mut conn, krate).await?
             );
             Ok(())
-        });
+        })
+        .await;
     }
 
-    #[test]
-    fn targets_default_to_one_with_timeout() {
-        wrapper(|env| {
-            let db = env.db();
+    #[tokio::test]
+    async fn targets_default_to_one_with_timeout() {
+        async_wrapper(|env| async move {
+            let db = env.async_db().await;
+            let mut conn = db.async_conn().await;
             let krate = "hexponent";
             Overrides::save(
-                &mut db.conn(),
+                &mut conn,
                 krate,
                 Overrides {
                     timeout: Some(Duration::from_secs(20 * 60)),
                     ..Overrides::default()
                 },
-            )?;
-            let limits = Limits::for_crate(&env.config(), &mut db.conn(), krate)?;
+            )
+            .await?;
+            let limits = Limits::for_crate(&env.config(), &mut conn, krate).await?;
             assert_eq!(limits.targets, 1);
 
             Ok(())
-        });
+        })
+        .await;
     }
 
-    #[test]
-    fn config_default_memory_limit() {
-        wrapper(|env| {
+    #[tokio::test]
+    async fn config_default_memory_limit() {
+        async_wrapper(|env| async move {
             env.override_config(|config| {
                 config.build_default_memory_limit = Some(6 * GB);
             });
 
-            let db = env.db();
+            let db = env.async_db().await;
+            let mut conn = db.async_conn().await;
 
-            let limits = Limits::for_crate(&env.config(), &mut db.conn(), "krate")?;
+            let limits = Limits::for_crate(&env.config(), &mut conn, "krate").await?;
             assert_eq!(limits.memory, 6 * GB);
 
             Ok(())
-        });
+        })
+        .await;
     }
 
-    #[test]
-    fn overrides_dont_lower_memory_limit() {
-        wrapper(|env| {
-            let db = env.db();
+    #[tokio::test]
```

Comment:
I haven't looked at the rest yet, but one thing I dislike about using `#[tokio::test]` is that we then aren't using the `ctx.runtime()`. We have some code like `utils::spawn_blocking` that relies on the ambient runtime while other code gets it directly from the context, so there may be some weird effects from having these different.

Reply:
That's definitely a valid point. 

What do you think is the better goal to have? Using the contrext runtime? Or tokio::main and tokio::test?

Reply:
If we can get rid of `Context::runtime` and ensure everywhere has an ambient runtime available, that's probably the best long-term plan. That has issues around `postgres` and `reqwest::blocking` code not being able to run anywhere within a runtime though.

Reply:
So this feels like we should keep `tokio::test` here, while preventing that there is a second runtime. Probably `ctx.runtime()` just returning the ambient runtime and never creating a new one? 

The short term alternative for this PR would probably be to make the test methods sync again, also make the `async_wrapper` sync, while internally using the ctx.runtime to `block_on`.

Reply:
I chose the simple approach in this simple first version, to not block this PR further. 

We won't start using `tokio::test` yet, I'll look into this in a later PR again.